test/e2e/storage/csi_volumes.go

```diff
@@ -237,4 +240,61 @@ var _ = utils.SIGDescribe("CSI Volumes [Flaky]", func() {
 			testDynamicProvisioning(t, cs, claim, class)
 		})
 	})
+
+	Describe("Sanity CSI plugin test using GCE-PD CSI driver", func() {
```

Comment:
Do you want to turn this into more of a framework, so you don't have to copy paste everything every time you want to test a new driver?  I see all of this new code almost copies the hostpath test, and just changes a few variables here and there.

Instead, each driver test can provide init/cleanup functions, and maybe some parameters for the test.

Reply:
hmmm. currently the two tests that exist seem different enough to me that they require their own seperate setup/teardown bits. Although they use the same functions they're doing different enough things that any "framework/configuration" that could accurately capture it might be almost as verbose as the config itself.

Especially since there are only two tests right now I don't want to prematurely generalize before we understand what other drivers might require.

Maybe i'm misunderstanding what you're asking for. Feel free to ping me if you have more thoughts.

Reply:
Yes, I mean each plugin still needs its own setup/teardown functions.  However, the actual test itself, the dynamic provisioning part, is common to the two tests.  I'm thinking this will make it easier in the future to add more test cases, like mount/unmount, read/write.

If you look at other test suites, like volume_provisioning, or subpath, they try to generalize the test case so it's easy to add more volume types to the test suite.